### PR TITLE
:godmode:

### LINKED ISSUE #13: :weary:
screen shot 2015-10-30 at 8 20 33 am

:rage3: @Scotchester @higs4281

### LINKED ISSUE #16: Update tests so they can be run locally
Instead of defining `TRAVIS_COMMIT_MSG` in a travis hook, define it directly in `test/index.js` so that the emoji test can be run locally.

See https://github.com/cfpb/CFPBot/pull/15#discussion-diff-43507668.

### PR DESCRIPTION
**Pull request by contolini**
Fixes #13 by checking commit messages against list of valid github markdown emoji.

Update: Fixes #16.

### CODE REVIEW

**Comment by Scotchester** [comment]
ROFL:helicopter:

**Comment by Scotchester** [comment]
I mean, it won't _fix_ #13. You want to rebase and fix that commit message before we merge this?

File: test/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  if (!error && response.statusCode == 200 && JSON.parse(emojis).some(isAnEmoji)) {
+    process.exit(0);
+  } else {
+    if (error) {
+      console.log(error);
+    }
+    console.error('Commit message is not a single emoji!');
+    process.exit(1);
+  }
+})
```
Comment (Scotchester) [inline test/index.js]: Missing semicolon.

**Comment by contolini** [comment]
@Scotchester I'm cool leaving it how it is. @higs4281 submitted #14 so that it's no longer showing in the GH UI.

File: test/index.js (hunk 1 above)
Comment (contolini) [inline test/index.js]: Fixed.

**Comment by Scotchester** [comment]
It still does if you go to https://github.com/cfpb/CFPBot/commits/master

**Comment by contolini** [comment]
@Scotchester Good point. Squashed it.

**Comment by Scotchester** [comment]
Problem:

```sh
------------------------------------------------------------------------------------------ 10:13:17
~/Projects/CFPBot contolini-verify-commits ⚓ node test
module.js:338
    throw err;
          ^
Error: Cannot find module 'request'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/cranfills/Projects/CFPBot/test/index.js:1:77)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
------------------------------------------------------------------------------------------ 10:13:26
~/Projects/CFPBot contolini-verify-commits ⚓
```

**Comment by contolini** [comment]
@Scotchester my bad. `npm i` and try again.

**Comment by Scotchester** [comment]
Oh, uh, just realized it was already listed in `dependencies`; I just hadn't run `npm install` at all. Do we still want to add it as a `devDependency`, or is that redundant?

**Comment by contolini** [comment]
Yeah, leave it to denote it's needed for testing. Whatever prod feature uses it might get removed some day.

File: test/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+var request = require('request'),
+    msg = process.env.TRAVIS_COMMIT_MSG;
```
Comment (Scotchester) [inline test/index.js]: Shouldn't this be something like

```js
msg = process.env.TRAVIS_COMMIT_MSG || <most recent local commit message>;
```

?

Comment (contolini) [inline test/index.js]: Ideally, yes. But it's not a quick one-liner. You'd have to spawn a child process to run the git command and that's more work than I wanted to do for this quick PR.

Comment (Scotchester) [inline test/index.js]: Makes sense. Should we update the filename/npm script name to make it clear that this test only runs in the context of Travis, then?

**Comment by Scotchester** [comment]
Also, we have another problem. The badge will never show passing, because as soon as a merge happens, Travis will run against the merge commit.

See: https://travis-ci.org/cfpb/CFPBot/builds

**Comment by contolini** [comment]
@Scotchester :expressionless:

**Comment by contolini** [comment]
@Scotchester The test is now rewritten so that it works locally. Travis is failing because it's testing master which has the bad emoji in it. I'm going to merge this and if Travis continues to fail we can start a separate issue.

**Comment by Scotchester** [comment]
So... what happened here, buddy? :)

**Pull request merged**